pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java

```diff
@@ -85,6 +90,18 @@ private String getDescription(Element element) {
 		return res;
 	}
 
+    private void validatePmd7Readiness(Element element) {
+        // PMD 7 expects rules to have a "language" property.
+        String langProp = element.getAttribute(ATTR_LANGUAGE);
+        // Rules can be deprecated.
+        boolean deprecated = element.getAttribute(ATTR_DEPRECATED).equalsIgnoreCase("true");
```

Comment:
I'm not sure we should exclude deprecated rules from this check. We won't know when they plan to remove them entirely. This warning could be a reason for them to delete it or update it if they think the rule should stay around longer.